EventListener/DestroyEventListener.php

```diff
@@ -57,39 +58,19 @@ public function onDestroy(DestroyEvent $event)
                 }
 
                 $manageSock = $this->dataValueService->getValue($server, 'manage_sock');
-
                 if (!$manageSock) {
                     continue;
                 }
 
-                $accountId = $this->dataValueService->getValue($applicationEnvironment, 'sock_account_id');
-
-                if ($accountId) {
-                    $this->apiService->removeAccount($accountId);
-                    $this->dataValueService->storeValue($applicationEnvironment, 'sock_account_id', null);
-                    $this->taskLoggerService->addLine(sprintf('Removed sock account %s.', $accountId));
-                }
-
-                $applicationId = $this->dataValueService->getValue($applicationEnvironment, 'sock_application_id');
-                if ($applicationId) {
-                    $this->apiService->removeApplication($applicationId);
-                    $this->dataValueService->storeValue($applicationEnvironment, 'sock_application_id', null);
-                    $this->taskLoggerService->addLine(sprintf('Removed sock application %s.', $applicationId));
-                }
-
-                $databaseId = $this->dataValueService->getValue($applicationEnvironment, 'sock_database_id');
-                if ($databaseId) {
-                    $this->apiService->removeDatabase($databaseId);
-                    $this->dataValueService->storeValue($applicationEnvironment, 'sock_application_id', null);
-                    $this->dataValueService->storeValue($applicationEnvironment, 'sock_database_id', null);
-                    $this->taskLoggerService->addLine(sprintf('Removed sock database %s.', $databaseId));
-                }
+                $this->destroySockDatabase($appEnv);
+                $this->destroySockApplication($appEnv);
+                $this->destroySockAccount($appEnv);
 
-                $applicationEnvironment->setDatabaseUser(null);
-                $applicationEnvironment->setDatabaseName(null);
-                $applicationEnvironment->setDatabasePassword(null);
+                $appEnv->setDatabaseUser(null);
```

Comment:
Shouldn't this go in destroy sock database? Or right after the call to the function destroySockDatabase?